src/onboarding/registration/NameAndPicture.tsx

```diff
@@ -84,8 +93,15 @@ function NameAndPicture({ navigation }: Props) {
       })
     }
   }
-
+  const onPressSkip = () => {
+    // TODO additional anlytics
+    handleNewName({ skipped: true })
+  }
   const onPressContinue = () => {
+    handleNewName({ skipped: false })
+  }
+
+  const handleNewName = ({ skipped }: { skipped: boolean }) => {
```

Comment:
It appears that if a user starts typing and then clicks skip, the name will still be saved. This is not expected behavior

Reply:
When you fix this, please also add a test.

Reply:
Thanks for catching it, this definitely needs a good change.

Reply:
I got a test added, but for some reason I cannot reproduce the bug; see my reply in this thread or #activation-eng

Reply:
in case anyone's reading along and curious about this^, we found that the name needed to be added as a dep for a react hook, so the hook would get evaluated again when the name changed